retrofit/src/test/java/retrofit2/RequestBuilderTest.java

```diff
@@ -2578,6 +2579,44 @@
     assertThat(readBody.indexOf("secondParam")).isLessThan(readBody.indexOf("thirdParam"));
   }
 
+
+  @Test public void queryParamsSkippedIfConvertedToNull() throws IOException, InterruptedException {
+    class Example {
+      @POST("/query") Call<String> queryPath(@Query("a") Object a) { return null; }
+    }
+
+    Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
+            .baseUrl("http://example.com")
+            .addConverterFactory(new NullObjectConverterFactory());
+
+    Request request = buildRequest(Example.class, retrofitBuilder, "Ignored");
+
+    if(request.url().toString().contains("Ignored")){
+      fail();
+    }
+  }
```

Comment:
I _think_ tests tend to have some assertion that succeeds to pass. Maybe assert that request.url().queryParameter("Ignored") is null?